python/covid_web/covid_web/rest.py

```diff
@@ -79,7 +82,34 @@ def get_app(self):
         return self
 
     def get_available_predictions(self):
-        return jsonify(self.available_predictions)
+        result = []
+        for prediction in self.available_predictions:
```

Comment:
Since available predictions are created in the constructor, why isn't the link set then?

Reply:
Links are not working at that stage. I got
`RuntimeError: Attempted to generate a URL without the application context being pushed. This has to be executed when application context is available.`

Reply:
Then I think we should do it differently, because this is inefficient. You create links every time you call this function.

- Make `_create_available_predictions` public, so it becomes `Rest.create_available_predictions()`. It will include link creation.
- Call it in `server.py` right before `def covid19_available_predictions()`.
- `def covid19_get_specific_prediction(prediction, country)` should be moved up in `server.py`, before `covid19_available_predictions`.

Reply:
Or there's another way: do it only once. Check if the available predictions are initialized and if not, initialize them.

Reply:
I seems that it's still too soon. It looks like `url_for` is available when Flask variable `SERVER_NAME` is set. This is done during execution, but it also can be done before it, by manually setting it. It seems to be quite hacky. I suggest just removing it (we can live without it) or creating it lazily once during the first call.

ref: https://stackoverflow.com/questions/31766082/flask-url-for-error-attempted-to-generate-a-url-without-the-application-conte

Reply:
I should perhaps also ask why this isn't instead stored offline and served via Nginx, as we plan to do at some point: https://github.com/lukipuki/COVID-19-simulation/issues/95

Reply:
Another PR.